src/main/java/javaslang/collection/Seq.java

```diff
@@ -198,6 +198,25 @@ default boolean containsSlice(java.lang.Iterable<? extends T> that) {
     }
 
     /**
+     * Tests whether every element of this sequence relates to the corresponding element of another sequence by satisfying a test predicate.
+     *
+     * @param that the other sequence
+     * @param predicate the test predicate, which relates elements from both sequences
+     * @return {@code true} if both sequences have the same length and {@code predicate(x, y)}
+     * is {@code true} for all corresponding elements {@code x} of this sequence and {@code y} of {@code that}, otherwise {@code false}.
+     */
+    default <U> boolean corresponds(java.lang.Iterable<U> that, Predicate<Tuple2<T, U>> predicate) {
```

Comment:
changing the signature as suggested above (? extends, ? super) then the following test does not compile any more:

```java
// error: operator '-' not applicable to Object
assertThat(of(1, 2, 3).corresponds(of(3, 4, 5), t -> t._1 == t._2 - 2)).isTrue();
```

Therefore I will leave the signature as you committed it.